app/models/configuration_source.rb

```diff
@@ -0,0 +1,6 @@
+class ConfigurationSource < ApplicationRecord
+  # An Tower project/repo has many playbooks
+  has_many    :configuration_script_payloads, dependent => :destroy
+  belongs_to  :manager
+  belongs_to  :scm_reference
```

Comment:
We'll need `repo type`, `url` and `branch`.  Check the Tower API for creating Tower Project - it doesn't have a `id` for a repo - it's not an entity in Tower. So if we choose not to model this in MIQ, we'll have to pass all 3 pieces of info every time.